src/ant/src/AntennaChecker.cc

```diff
@@ -740,6 +740,92 @@ bool AntennaChecker::checkCSR(odb::dbTechLayer* tech_layer,
   return violation;
 }
 
+bool AntennaChecker::checkRatioViolations(odb::dbTechLayer* layer,
+                                          const NodeInfo& node_info,
+                                          bool verbose,
+                                          bool report,
+                                          std::ofstream& report_file)
+{
+  bool node_has_violation = false;
+  auto par_violation = checkPAR(layer, node_info, verbose, report, report_file);
+  bool car_violation = checkCAR(layer, node_info, verbose, report, report_file);
+  node_has_violation = par_violation.first || car_violation;
```

Comment:
There's no need to define node_has_violation and then set it afterwards, just combine 749 & 752